server/game/cards/03_TWI/upgrades/AhsokasPadawanLightsaber.ts

```diff
@@ -0,0 +1,31 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { Card } from '../../../core/card/Card';
+import { Trait } from '../../../core/Constants';
+import { UpgradeCard } from '../../../core/card/UpgradeCard';
+
+export default class AhsokasPadawanLightsaber extends UpgradeCard {
+    protected override getImplementationId() {
+        return {
+            id: '0741296536',
+            internalName: 'ahsokas-padawan-lightsaber'
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.setAttachCondition((card: Card) => !card.hasSomeTrait(Trait.Vehicle));
+
+        this.addWhenPlayedAbility({
+            title: 'Attack with a unit',
+            optional: true,
+            targetResolver: {
+                immediateEffect: AbilityHelper.immediateEffects.conditional((context) => ({
```

Comment:
Instead of using a context handler for the whole immediateEffect, you could just use one for conditional like `condition: (context) => context.source.parentCard?.title === 'Ahsoka Tano'`

Reply:
good deal, fixed up